src/build-tools/src/lib.rs

```diff
@@ -68,3 +68,55 @@ pub fn protoc_include() -> PathBuf {
         }
     }
 }
+
+/// The current git hash.
+pub const GIT_HASH: &str = crate::git_hash();
+
+/// Returns the current `git` hash.
+///
+/// Identifies the current hash in one of two ways:
+///
+/// 1. Relies on the `:git_version` Bazel rule generating a Rust source file with the hash.
+/// 2. Uses [`compile_time_run`] to generate the hash at compile time.
+///
+pub const fn git_hash() -> &'static str {
```

Comment:
I think this needs to be a macro, not a function, for the reason listed in the docstring on the `build_info!` macro:

https://github.com/MaterializeInc/materialize/pull/27411/files#diff-5f1a1ad1b67d9efd041c352c0bd727280421c330b55ff65fb371bd055c336cb2R78-R81

As written, `build-tools` will get compiled once by Cargo and essentially never recompiled, and so changes to other crates that depend on `build-tools` won't cause `build-tools` to get recompiled, and the commit SHA that gets embedded will frequently be stale.

Reply:
Actually @ParkMyCar I feel like this might all just belong directly in `mz-build-info`? Is there some reason that wouldn't work? I don't mind at all having Bazel goop in that crate.

Reply:
Ahhh yeah that makes sense, I'll update this!